api/v2/api.go

```diff
@@ -389,6 +395,89 @@ func (api *API) postAlertsHandler(params alert_ops.PostAlertsParams) middleware.
 	return alert_ops.NewPostAlertsOK()
 }
 
+func (api *API) getAlertGroupsHandler(params alertgroup_ops.GetAlertGroupsParams) middleware.Responder {
+	var (
+		err            error
+		receiverFilter *regexp.Regexp
+		matchers       = []*labels.Matcher{}
+	)
+
+	for _, matcherString := range params.Filter {
+		matcher, err := parse.Matcher(matcherString)
+		if err != nil {
+			level.Error(api.logger).Log("msg", "failed to parse matchers", "err", err)
+			return alertgroup_ops.NewGetAlertGroupsBadRequest().WithPayload(err.Error())
+		}
+
+		matchers = append(matchers, matcher)
+	}
+
+	if params.Receiver != nil {
+		receiverFilter, err = regexp.Compile("^(?:" + *params.Receiver + ")$")
+		if err != nil {
+			return alertgroup_ops.
+				NewGetAlertGroupsBadRequest().
+				WithPayload(
+					fmt.Sprintf("failed to parse receiver param: %v", err.Error()),
+				)
+		}
+	}
+
+	alertGroups := api.groups(matchers, receiverFilter, *params.Silenced, *params.Inhibited, *params.Active)
+
+	res := make(open_api_models.AlertGroups, 0, len(alertGroups))
+
+	for _, alertGroup := range alertGroups {
+		ag := &open_api_models.AlertGroup{
+			Receiver: alertGroup.Receiver,
+			Labels:   modelLabelSetToAPILabelSet(alertGroup.Labels),
+			Alerts:   make([]*open_api_models.GettableAlert, 0, len(alertGroup.Alerts)),
+		}
+
+		for _, ea := range alertGroup.Alerts {
```

Comment:
What does `ea` stand for?

Reply:
enriched alert. happy to change it, it's fairly cryptic.

Reply:
That would be helpful, thanks.